CHANGELOG.md

```diff
@@ -1,5 +1,68 @@
 # Changelog
 
+## 0.0.1-alpha.14
+
+### Bug fixes
+
+- Fix descriptor lookups for most types that overlap with `None` ([#19120](https://github.com/astral-sh/ruff/pull/19120)).
+    This means that e.g. `object().__str__()` now correctly binds the `self` argument of the `__str__`
+    method, as the `object` type overlaps with `None`.
+- Don't allow first-party code to shadow the stdlib `types` module ([#19128](https://github.com/astral-sh/ruff/pull/19128)).
+    This fixes a possible source of stack overflows.
+- Add cycle detection to our ty's implementation of disjointness, preventing another possible source of stack overflows when analysing recursive types ([#19139](https://github.com/astral-sh/ruff/pull/19139))
+
+### Server
+
+- Filter a symbol from a stub file in autocomplete suggestions if it is an implementation detail of the stub ([#19121](https://github.com/astral-sh/ruff/pull/19121))
+
+- Add initial support for [semantic tokens](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens). This feature allows editors to apply more advanced syntax highlighting. ([#19108](https://github.com/astral-sh/ruff/pull/19108)).
+
+    Currently, the supported tokens are:
+
+    - `Namespace`
+    - `Class`
+    - `Parameter`
+    - `SelfParameter`
+    - `ClsParameter`
+    - `Variable`
+    - `Property`
+    - `Function`
+    - `Method`
+    - `Keyword`
+    - `String`
+    - `Number`
+    - `Decorator`
+    - `BuiltinConstant`
+    - `TypeParameter`
+
+- Initial support for [workspace diagnostics](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_diagnostic) ([#18939](https://github.com/astral-sh/ruff/pull/18939))
+
+- Use Python syntax highlighting for Markdown code fences in on-hover content ([#19082](https://github.com/astral-sh/ruff/pull/19082))
+
+### Other changes
+
+- Add subtyping between `type[]` types and `Callable` types ([#19026](https://github.com/astral-sh/ruff/pull/19026))
+- Support bare `ClassVar` annotations ([#15768](https://github.com/astral-sh/ruff/pull/15768))
+- Understand that calls to functions returning `Never` / `NoReturn` are terminal with respect to control flow ([#18333](https://github.com/astral-sh/ruff/pull/18333))
```

Comment:
In my opinion, this deserves to be higher up. It's a fantastic new feature that removes many false positives.

Reply:
moved it to be the top item in the "typing semantics" section